### PR TITLE
Create custom date picker calendar

### PR DESCRIPTION
**Pull request by ghulamhafeez**
### Implementation

- Create a custom date picker, include years and months selection options within the calendar component
- Also implement on ehad duration & date of birth fields

### Resources

https://github.com/user-attachments/assets/e8f22173-11a6-49fc-a787-56e6dd4e6ff5

### CODE REVIEW

File: src/components/ui/datepicker.tsx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+interface DatePickerProps<T extends FieldValues> {
+  startYear?: number;
+  endYear?: number;
+  label: string;
+  control: Control<T>;
+  name: FieldPath<T>;
+}
+
+export function DatePicker<T extends FieldValues>({
+  startYear = getYear(new Date()) - 60,
```
Comment (muhammad-bilal-ali-khokhar) [inline src/components/ui/datepicker.tsx:39]: i think you no need this prop `startyear` and `endYear` you can direclty use in this component

File: src/components/ui/datepicker.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  FormMessage,
+} from "./form";
+import { Control, FieldPath, FieldValues } from "react-hook-form";
+
+interface DatePickerProps<T extends FieldValues> {
+  startYear?: number;
+  endYear?: number;
+  label: string;
+  control: Control<T>;
+  name: FieldPath<T>;
```
Comment (muhammad-bilal-ali-khokhar) [inline src/components/ui/datepicker.tsx:35]: why you are using generic type parameter in `control` and `name`
and one question is, name is not a string?

File: src/components/ui/datepicker.tsx [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+}
+
+export function DatePicker<T extends FieldValues>({
+  startYear = getYear(new Date()) - 60,
+  endYear = getYear(new Date()),
+  label,
+  control,
+  name,
+}: DatePickerProps<T>) {
```
Comment (muhammad-bilal-ali-khokhar) [inline src/components/ui/datepicker.tsx:44]: I don't think we need generic type `[T]`

File: src/components/ui/datepicker.tsx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+          field.onChange(newDate);
+        };
+
+        const handleYearChange = (year: string) => {
+          const newDate = setYear(date, parseInt(year));
+          setDate(newDate);
+          field.onChange(newDate);
+        };
+
+        const handleSelect = (selectedDate: Date | undefined) => {
```
Comment (muhammad-bilal-ali-khokhar) [inline src/components/ui/datepicker.tsx:83]: how selected date can be undefined

File: src/components/ui/datepicker.tsx (hunk 1 above)
Comment (ghulamhafeez) [reply src/components/ui/datepicker.tsx:39]: Okay Fixed

**Review by Mubashar1009 — APPROVED** [review]

**Pull request merged**